include/nlohmann/detail/conversions/from_json.hpp

```diff
@@ -107,9 +107,11 @@ void from_json(const BasicJsonType& j, typename BasicJsonType::string_t& s)
 template <
     typename BasicJsonType, typename ConstructibleStringType,
     enable_if_t <
-        is_constructible_string_type<BasicJsonType, ConstructibleStringType>::value&&
-        !std::is_same<typename BasicJsonType::string_t,
-                      ConstructibleStringType>::value,
+        is_constructible_string_type<BasicJsonType, ConstructibleStringType>::value
```

Comment:
Do we still need the `is_constructible_string_type` check since we do assignment, or are we also doing construction somewhere?

Reply:
Good question. I think we can remove that.